src/common/lru_cache.h

```diff
@@ -356,6 +356,195 @@ std::shared_ptr<CacheMetrics>
+class SglLRUCache : public SglLRUCacheInterface<K> {
+ public:
+    explicit SglLRUCache(uint64_t maxCount,
+        std::shared_ptr<CacheMetrics> cacheMetrics = nullptr)
+      : maxCount_(maxCount),
+        cacheMetrics_(cacheMetrics) {}
+
+    explicit SglLRUCache(std::shared_ptr<CacheMetrics> cacheMetrics = nullptr)
+      : maxCount_(0),
+        cacheMetrics_(cacheMetrics) {}
+
+    void Put(const K &key) override;
+
+    bool IsCached(const K &key) override;
+
+    void Remove(const K &key) override;
+
+    void GetBack(K *value);
+
+    uint64_t Size();
+
+    std::shared_ptr<CacheMetrics> GetCacheMetrics() const;
+
+ private:
+    bool PutLocked(const K &key);
+
+    void RemoveLocked(const K &key);
+
+    void MoveToFront(const typename std::list<K>::iterator &elem);
+
+    bool RemoveOldest();
+
+    void RemoveElement(const typename std::list<K>::iterator &elem);
+
+ private:
+    ::curve::common::RWLock lock_;
+
+    // the maximum length of the queue. 0 indicates unlimited length
+    uint64_t maxCount_;
+    // dequeue for storing items
+    std::list<K> ll_;
+    // record the position of the item corresponding to the key in the dequeue
+    std::unordered_map<K, typename std::list<K>::iterator> cache_;
+
+    // cache related metric data
+    std::shared_ptr<CacheMetrics> cacheMetrics_;
+};
+
+template <typename K, typename KeyTraits>
+std::shared_ptr<CacheMetrics>
+    SglLRUCache<K, KeyTraits>::GetCacheMetrics() const {
+    return  cacheMetrics_;
+}
+
+template <typename K, typename KeyTraits>
+uint64_t SglLRUCache<K, KeyTraits>::Size() {
+    ::curve::common::WriteLockGuard guard(lock_);
+    return ll_.size();
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::Put(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    PutLocked(key);
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::GetBack(K *value) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    if (ll_.empty()) {
+        LOG(INFO) << "cache is empty.";
+        return;
+    }
+    *value = ll_.back();
+    return;
+}
+
+template <typename K, typename KeyTraits>
+bool SglLRUCache<K, KeyTraits>::IsCached(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    auto iter = cache_.find(key);
+    if (iter == cache_.end()) {
+        if (cacheMetrics_ != nullptr) {
+            cacheMetrics_->OnCacheMiss();
+        }
+        return false;
+    }
+
+    if (cacheMetrics_ != nullptr) {
+        cacheMetrics_->OnCacheHit();
+    }
+
+    // update the position of the target item in the list
+    MoveToFront(iter->second);
+    return true;
+}
+
+template <typename K, typename KeyTraits>
+void SglLRUCache<K, KeyTraits>::Remove(const K &key) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    RemoveLocked(key);
+}
+
+template <typename K, typename KeyTraits>
+bool SglLRUCache<K, KeyTraits>::PutLocked(const K &key) {
```

Comment:
If you don't care about the evicted item, returning void is enough.

Reply:
fix